File: src/main/java/de/unistuttgart/iste/rss/tutormanagementsystem/model/scheincriteria/ScheinCriteriaDTO.java
package de.unistuttgart.iste.rss.tutormanagementsystem.model.scheincriteria;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.Getter;

/**
 * CreateScheinCriteriaDTO
 */
@Getter
public class ScheinCriteriaDTO {

    private String identifier;
    private JsonNode data;

}

File: server/src/module/scheincriteria/scheincriteria.dto.ts
import { IsDefined, IsString } from 'class-validator';
import { IScheinCriteriaDTO } from 'shared/model/ScheinCriteria';

export class ScheinCriteriaDTO implements IScheinCriteriaDTO {
    @IsString()
    name!: string;

    @IsString()
    identifier!: string;

    @IsDefined()
    data: any;
}
